rune/libenclave/internal/runtime/pal/pal_linux.go

```diff
@@ -3,27 +3,67 @@ package enclave_runtime_pal // import "github.com/inclavare-containers/rune/libe
 import (
 	"encoding/binary"
 	"fmt"
+	"github.com/alibaba/inclavare-containers/epm/pkg/epm-api/v1alpha1"
 	"github.com/go-restruct/restruct"
 	"github.com/inclavare-containers/rune/libenclave/attestation"
 	"github.com/inclavare-containers/rune/libenclave/attestation/sgx"
 	_ "github.com/inclavare-containers/rune/libenclave/attestation/sgx/ias"
+	"github.com/inclavare-containers/rune/libenclave/epm"
 	"github.com/inclavare-containers/rune/libenclave/intelsgx"
+	"strings"
+
 	"log"
 	"os"
 )
 
 const (
-	palApiVersion = 3
+	palApiVersion        = 3
+	InvalidEpmID  string = "InvalidEPMID"
 )
 
 func (pal *enclaveRuntimePal) Init(args string, logLevel string) error {
+	var enclaveinfo *v1alpha1.Enclave
+	var err error
+	var addr int64 = 0
+	var fd int = -1
+
+	pal.enclavePoolID = InvalidEpmID
 	api := &enclaveRuntimePalApiV1{}
 	ver := api.get_version()
 	if ver > palApiVersion {
 		return fmt.Errorf("unsupported pal api version %d", ver)
 	}
 	pal.version = ver
 
+	/* FIXME: If EPM provides epm existence detect API, the static check will be substituted. Enclave pool
+	 * will be distinguished by pal.Type and Pal.subType once subType can be provided by new PAL interface
+	 * in future.
+	 */
+	if strings.Contains(args, "epm") {
+		if strings.Contains(args, "skeleton") {
+			pal.Type = "enclave-cache.skeleton"
+		} else if strings.Contains(args, "occlum") {
+			pal.Type = "enclave-cache.occlum"
+		}
```

Comment:
Because this layer doesn't wanna know any details about enclave runtime itself, so we can use the key word "mock" as the demonstration purpose, instead of listing skeleton and occult.

Reply:
In epm server, the enclave pool type has been set for "enclave-cache.skeleton"  and "enclave-cache.occlum". And it has been merged.  In config.json, enclave.runtime.args includes skeleton when running skeleton instance, so strings.Contains(args, "skeleton") shoud be reasonable. What do you think？ 
If you think "enclave-cache.occlum" is not good demonstration, we can replace it by constant string type, e.g. SkeletonEnclaveType string  =  "enclave-cache.skeleton"

Reply:
Specifying "skeleton" in args doesn't make sense. It just a place holder without any effect. Sorry that it misleads you. I will open a new issue to remove it.

In other words, rune doesn't need to know what the enclave runtime is exactly. Unlike shim-rune which requires converting bundle, rune doesn't have to know it.

Reply:
I got you.  Thanks